compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -9065,7 +9073,18 @@ private STNode parseResourceAccessSegmentRhs(boolean isRhsExpr, boolean isInMatc
                 return parseResourceAccessSegmentRhs(isRhsExpr, isInMatchGuard);
         }
     }
-    
+    private STNode parseOnFailCheck() {
```

Comment:
Let's document the grammar parsed by this method smilar to other methods. Also, need to update `parseCheckExpression()` doc to reflect the new grammar.  e.g. `checking-expr := checking-keyword expression [on-fail-check]`

Reply:
Updated